modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -0,0 +1,318 @@
+<?php
+/**
+ * Jetpack_WooCommerce_Analytics_Universal
+ *
+ * @package Jetpack
+ * @author Automattic
+ */
+
+/**
+ * Bail if accessed directly
+ */
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Jetpack_WooCommerce_Analytics_Universal
+ * Filters and Actions added to Store pages to perform analytics
+ */
+class Jetpack_WooCommerce_Analytics_Universal {
+	/**
+	 * Jetpack_WooCommerce_Analytics_Universal constructor.
+	 */
+	public function __construct() {
+
+		// loading s.js
+		add_action( 'wp_head', array( $this, 'wp_head' ), 999999 );
+
+		// single product page view
+		add_action( 'woocommerce_after_single_product', array( $this, 'product_detail' ) );
+
+		// add to cart on single product page
+		add_action( 'woocommerce_after_add_to_cart_button', array( $this, 'add_to_cart' ) );
+
+		// add to carts from list views (search, store etc.)
+		add_action( 'wp_footer', array( $this, 'loop_add_to_cart' ) );
+
+
+		add_action( 'woocommerce_after_cart', array( $this, 'remove_from_cart' ) );
+		add_action( 'woocommerce_after_mini_cart', array( $this, 'remove_from_cart' ) );
+		add_action( 'wcct_before_cart_widget', array( $this, 'remove_from_cart' ) );
+		add_filter( 'woocommerce_cart_item_remove_link', array( $this, 'remove_from_cart_attributes' ), 10, 2 );
+
+		// cart checkout
+		add_action( 'woocommerce_after_checkout_form', array( $this, 'checkout_process' ) );
+
+		// order confirmed
+		add_action( 'woocommerce_thankyou', array( $this, 'order_process' ), 10, 1 );
+		add_action( 'woocommerce_after_cart', array( $this, 'remove_from_cart_via_quantity' ), 10, 1 );
+	}
+
+	/**
+	 * Place script to call s.js, Store Analytics
+	 */
+	public function wp_head() {
+
+		// If we're in the admin_area, return without inserting code.
+		if ( is_admin() ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WooCommerce' ) ) {
+			return;
+		}
+
+		$async_code = "<script async src='https://stats.wp.com/s.js'></script>";
+
+		echo "$async_code\r\n";
+	}
+
+	/**
+	 * On a product page, add a click event listener to "Add to Cart" button click
+	 */
+	public function add_to_cart() {
+
+		if ( ! is_single() ) {
+			return;
+		}
+
+		$blogid = Jetpack::get_option( 'id' );
+		global $product;
+
+		$selector = '.single_add_to_cart_button';
+
+		wc_enqueue_js(
+			"window._wca = window._wca || [];
+			jQuery( '" . esc_js( $selector ) . "' ).click( function() {
```

Comment:
@greenafrican Clicking the "Add to Cart" button here cause a page navigation. This may cause an issue with a page navigating before our event is triggered.